Games/types/Mafia/roles/Mafia/Apprentice.js

```diff
@@ -0,0 +1,21 @@
+const Role = require("../../Role");
+
+module.exports = class Apprentice extends Role {
+
+    constructor(player, data) {
+        super("Apprentice", player, data);
+
+        this.alignment = "Mafia";
+        this.cards = ["VillageCore", "WinWithMafia", "MeetingMafia", "BecomeDeadRole"];
+        this.meetingMods = {
+            "Become Role": {
+                targets: { include: [executedMafia], exclude: ["alive", "self"] },
+            }
+        };
+    }
+
+}
+
+function executedMafia(player) {
+    return player.deathType == "lynch" && player.role.alignment == "Mafia";
```

Comment:
player.deathType is not a real function

when people die this information is currently not being recorded

see `Player::kill(killType, killer, instant)`

you will need to make a listener and add the player

While it is tempting to set player.deathType, that is not right since each phase allows a player to die multiple times. E.g. player who is both poisoned AND voted off, or vegges and is voted off.

Apprentice is an underwhelming role as is, I'm fine if it can remember all dead mafia.